### DISCUSSION TITLE
!Translating the App: A Step-by-Step Guide

### DISCUSSION THREAD

**Discussion by DasFragezeichen** [builtbybel/xd-AntiSpy#12] [General]
German Interface / Translation would be nice.

**Comment by Mira-Bellenbaum** [comment]
Belim antwortete bereits im anderen Zweig auf eine diesbezügliche Frage.
"... ich erinnere mich, dass es ... eine deutsche Resourcedatei gab, als die App noch in C++ geschrieben war.
... ich sie mal neu Lokalisieren (den Code muss ich sowieso neu schreiben) . Die Pflege der deutschen Übersetzung würde ich allerdings gerne abgeben. ..."

Also abwarten!

**Comment by Belim** [comment]
With the release of 4.0.6, the app is now "localizable". I’ve also created an initial AI-generated German language file. If you would like to adjust and maintain it in the future, please follow the instructions below 😸 

1. Grab the tool from https://github.com/HakanL/resxtranslator/releases. This handy tool lets non-developers swiftly translate .NET project resource files (resx) into multiple languages simultaneously.
2. Download the language files in resx format from [here](https://github.com/builtbybel/xd-AntiSpy/tree/main/LocalizationLibrary/Locales). It's best to grab the entire raw "locales" folder: 
3. Open ResxTranslator.exe, navigate to File > Open directory, and locate the previously downloaded "locales" folder. The tool will scan all subfolders to find resx files (note: it doesn't rely on VS SLN or project files, just files with the .resx extension).
4. In the left panel, you'll see a tree view of all discovered resources. Double-click a node to open a resource and view its strings on the right side of the screen. 
5. Once you've translated all strings, go to File > Export all resources. Make sure to include a language code; for example, use "de" for German translations.
6. Send me the exported file by opening a new GitHub [Issue](https://github.com/builtbybel/xd-antispy/issues/new/choose) or via a new [Pull request](https://github.com/builtbybel/xd-antispy/pulls).

**Feel free to immortalize yourselves in the About Window (with name, URL, email). Use the area I've marked next to my name for this.**
![credits](https://github.com/user-attachments/assets/eefd711b-66c6-45eb-967f-cdd2141a1136)

**Comment by dai** [comment]
I was an XP-Antispy user. 
Thank you for your continued development.

and I've created a (ja-JP, Japanese) Japanese locale using the above procedure.

Best Regards.

![image](https://github.com/builtbybel/xd-AntiSpy/assets/12391/83046d60-84c0-44a5-a297-6413507e21a3)

**Comment by Mira-Bellenbaum** [comment]
@Belim 
Das mit dem Tool verstehe ich. Denke werde das hinbekommen.
Was mir aber aufgefallen ist, dass ab "Tweaks (Plugins)" die Keys, sowie die dazugehörigen englischen Texte fehlen.
AH, _catTaskbar habe ich gefunden, ergo muss ich _catTweaks (Plugins) neu erstellen.
Die anderen "Einträge" werde ich auch noch herausfinden, denke ich.

**Comment by Mira-Bellenbaum** [comment]
Was fehlt!

Optimierungen (Plugins)


Icon-Cache löschen

Diese Funktion löscht den Symbol-Cache in Windows, um Probleme mit der fehlerhaften Anzeige von Symbolen zu beheben.


Versteckte Dateierweiterungen ausblenden   ???????????? einblenden !!!!!!!!!!!!

Diese Funktion zeigt oder verbirgt versteckte Dateierweiterungen im Windows Explorer.


Snap Assist Flyout aktivieren

Damit wird das Snap Asssist Flyout in Windows 11 aktiviert.


Helles Thema verwenden ??? Theme !!!!!!!

Aktivieren Sie das Kontrollkästchen für den hellen Modus, deaktivieren Sie es für den dunklen Modus.


Sicherheit (Plugins)


Benutzerkontensteuerung aktivieren (UAC)

Aktivieren Sie diese Option, um die Benutzerkontensteuerung (UAC) zu aktivieren, deaktivieren Sie sie, um die UAC zu deaktivieren. Bitte deaktivieren Sie diese Einstellung nur, wenn Sie absolut sicher sind, dass Sie die vollständige Kontrolle über Ihr System haben!!! Andernfalls könnte dies eine große Sicherheitslücke in Ihrem System öffnen.

**Comment by Belim** [comment]
@Mira-Bellenbaum  Die Plugin-Strings sind nicht hardcodiert in der Anwendung, sondern befinden sich außerhalb des Kerns der Anwendung. Ich bin unsicher, ob es sinnvoll ist, diese zu übersetzen. Im Release habe ich auch angedeutet, dass die JSON-Dateien, also die Plugins und der toolDebloater, nicht von mir übersetzt wurden.

**Comment by Mira-Bellenbaum** [comment]
Mit ....
Enable Snap Assist Flyout

This will enable Snap Asssist Flyout in Windows 11

kann ich leider nichts anfangen, ich nutze kein Windows 11 und habe also keine Ahnung was für eine Funktion das ist.

**Comment by Mira-Bellenbaum** [comment]
Zitat: "@Mira-Bellenbaum Die Plugin-Strings sind nicht hardcodiert ...."
Ah, OK. Dann muss ich mich damit ja nicht weiter beschäftigen.

**Comment by Belim** [comment]
> I was an XP-Antispy user. Thank you for your continued development.
> 
> and I've created a (ja-JP, Japanese) Japanese locale using the above procedure.
> 
> Best Regards.
> 
> ![image](https://private-user-images.githubusercontent.com/12391/347266056-83046d60-84c0-44a5-a297-6413507e21a3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA2MDY2ODUsIm5iZiI6MTcyMDYwNjM4NSwicGF0aCI6Ii8xMjM5MS8zNDcyNjYwNTYtODMwNDZkNjAtODRjMC00NGE1LWEyOTctNjQxMzUwN2UyMWEzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzEwVDEwMTMwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYwZTg0YjdlM2I2Y2VjNzQ5NmQzMWNhZjdiZjY5NzBlNDQ0NzZiZThkODIzY2YzOTQ4OGY3OGIzODc3YTYyYzEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.5wJKjRk01i7wWkFB1w2qr6Sjn4Ksl3TW0R0r-o8ZgeI)

Cool, thanks. When you're done, please read through step 6 again or upload the file here.

**Reply by dai** [reply]
@Belim Thanks, I will start the translation now.

I'll send you a PR when the translation is complete.

**Comment by DMmoinmoin** [comment]
@Mira-Bellenbaum 
Enable Snap Assist Flyout ist das Andock Layout 

https://www.deskmodder.de/wiki/index.php?title=Windows_11_Maximieren_Schaltfläche_Andock_Layout_deaktivieren_aktivieren

**Comment by Camelot631** [comment]
.....
> 3. Open ResxTranslator.exe, navigate to File > Open directory, and locate the previously downloaded "locales" folder. The tool will scan all subfolders to find resx files (note: it doesn't rely on VS SLN or project files, just files with the .resx extension).

Hallöle...

Bis jetzt nur mitgelesen (...auch auf Deskmodder).....und jetzt am Testen... ;)

Bei mir erscheint eine Fehlermeldung beim Öffnen des Ordners...

Siehe Bild

Grüsse

Camelot631

<img width="674" alt="xdAntiSpy_German_20240710" src="https://github.com/builtbybel/xd-AntiSpy/assets/49121913/8612b0bc-98d9-4937-b5b4-2cc7e014ba2e">

**Comment by Belim** [comment]
@Camelot631 Am besten mal die Dateien erneut herunterladen, hier auch nochmal beigefügt
[Strings-de.zip](https://github.com/user-attachments/files/16164856/Strings-de.zip)

**Reply by Camelot631** [reply]
@Belim....

Danke Dir. Werde es heute nochmals testen...

**Kleiner Nachtrag...funzt jetzt wunderbar.... DANKE..... :)**
